File: backend/enrollment-service/src/main/java/com/lms/enrollmentservice/service/EnrollmentService.java
package com.lms.enrollmentservice.service;

import com.lms.enrollmentservice.dto.EnrollmentRequest;
import com.lms.enrollmentservice.dto.EnrollmentResponse;
import com.lms.enrollmentservice.dto.ProgressUpdateRequest;
import java.util.List;

public interface EnrollmentService {
    EnrollmentResponse createEnrollment(EnrollmentRequest request, Long studentId, String role);
    EnrollmentResponse getEnrollmentById(Long id, Long userId, String role);
    List<EnrollmentResponse> getEnrollmentsByStudent(Long studentId, Long authenticatedUserId, String role);
    List<EnrollmentResponse> getEnrollmentsByCourse(Long courseId, String role);
    EnrollmentResponse updateProgress(Long id, ProgressUpdateRequest request, Long userId, String role);
    EnrollmentResponse cancelEnrollment(Long id, Long userId, String role);
}

File: frontend/src/services/enrollmentService.js
import API from "./api";

const enrollmentService = {
  createEnrollment: async (courseId) => {
    const response = await API.post("/api/enrollments", { courseId });
    return response.data;
  },

  getEnrollmentById: async (id) => {
    const response = await API.get(`/api/enrollments/${id}`);
    return response.data;
  },

  getEnrollmentsByStudent: async (studentId) => {
    const response = await API.get(`/api/enrollments/student/${studentId}`);
    return response.data;
  },

  getEnrollmentsByCourse: async (courseId) => {
    const response = await API.get(`/api/enrollments/course/${courseId}`);
    return response.data;
  },

  updateProgress: async (id, progressPercentage) => {
    const response = await API.put(`/api/enrollments/${id}/progress`, {
      progressPercentage,
    });
    return response.data;
  },

  cancelEnrollment: async (id) => {
    const response = await API.put(`/api/enrollments/${id}/cancel`);
    return response.data;
  },
};

export default enrollmentService;
